settings.gradle

```diff
@@ -23,6 +23,7 @@ module('binary-compatibility-validator')
 module('common/kotlinx-coroutines-core-common')
 
 module('core/kotlinx-coroutines-core')
+module('core/kotlinx-coroutines-core/service-loader-test')
```

Comment:
Why a separate module is required?
Current tests can be safely moved to kx-coroutines-core.

It is also worth to add a special service to core resources, but write a test that loads this service from `android-unit-tests` module to test cross-module loading.

In that case, you can get rid of `public` + `InternalCoroutinesApi`and additional module

Reply:
For a single cross-module test you can use `@Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")`.
It is better to suppress the test then exposing something in the API

Reply:
I created that separate module to add dependencies to ```kotlinx-coroutines-test (android)``` modules. Now if I move those tests to ```kotlinx-coroutines-core``` adding dependencies to these modules then FailFastOnStart tests would try to load providers from those modules and fail.

Reply:
`FailFastOnStart` loads only `MainDispatcherFactory`.
You can use another service for tests.